internal/kibana/alerting.go

```diff
@@ -47,7 +48,7 @@ func ResourceAlertingRule() *schema.Resource {
 		"notify_when": {
 			Description:  "Defines how often alerts generate actions. Valid values include: `onActionGroupChange`: Actions run when the alert status changes; `onActiveAlert`: Actions run when the alert becomes active and at each check interval while the rule conditions are met; `onThrottleInterval`: Actions run when the alert becomes active and at the interval specified in the throttle property while the rule conditions are met. NOTE: This is a rule level property; if you update the rule in Kibana, it is automatically changed to use action-specific `notify_when` values.",
 			Type:         schema.TypeString,
-			Required:     true,
+			Optional:     true,
```

Comment:
This became optional because the preferred way is to pass the value inside the `frequency` in `actions`.

Reply:
Should we do the same for the `throttle`? It seems both became optional in 8.6+ in this [PR](https://github.com/elastic/kibana/pull/144130). Also, what will happen to <=8.5 if the fields are optional in TF but required in Kibana?

Reply:
> Also, what will happen to <=8.5 if the fields are optional in TF but required in Kibana?

I'm not sure how to handle that here, I can specify they are either optional or not — not depending on the version.

Maybe the documentation can specify that they are optional only from `8.6`?

Reply:
> Maybe the documentation can specify that they are optional only from 8.6

There's similar notes in the docs for other resources. 

It would also be a good idea to apply this validation within the resource code itself, i.e throw an error if create/update is called against ES<=8.5 and `notify_when` is not defined.

Reply:
I would also suggest throwing an error if possible on the resource level. @tobio Do you have an example of another resource that achieves the same?

Reply:
> I would also suggest throwing an error if possible on the resource level.

I already do something similar for `alert_delay` [here](https://github.com/elastic/terraform-provider-elasticstack/blob/main/internal/kibana/alerting.go#L195).